packages/data-point-service/lib/cache-middleware.js

```diff
@@ -244,40 +272,63 @@ function before (service, ctx, next) {
  * @param {Function} next Middleware callback
  */
 function after (service, ctx, next) {
-  const cache = getCacheParams(ctx.context.params)
+  const { generateKey, setStaleWhileRevalidateEntry } = module.exports
 
-  if (!cache.ttl) {
+  const cache = EntityCacheParams.getCacheParams(ctx.context.params)
+
+  // do nothing if:
+  // - cache is not configured, or
+  // - its at the process of revalidating, then lets skip any further calls
+  if (!cache.ttl || ctx.locals.revalidatingCache) {
     // do nothing
-    return next()
+    next()
+    return false
   }
 
   // from here below ttl is assumed to be true
   const entryKey = generateKey(cache.cacheKey, ctx)
 
-  if (cache.useStaleWhileRevalidate) {
-    // if its at the process of revalidating, then lets skip any further calls
-    if (ctx.locals.revalidatingCache) {
-      return next()
-    }
+  let resolution
 
+  if (cache.useStaleWhileRevalidate) {
     // adds (or updates) the stale cache entry with the latest value
-    return module.exports
-      .setStaleWhileRevalidateEntry(service, entryKey, ctx.value, cache)
-      .then(() => {
-        next()
-      })
+    resolution = setStaleWhileRevalidateEntry(
+      service,
+      entryKey,
+      ctx.value,
+      cache
+    ).then(() => {
+      next()
+    })
+  } else {
+    // adds a cache entry
+    resolution = RedisController.setEntry(
+      service,
+      entryKey,
+      ctx.value,
+      cache.ttl
+    )
   }
 
-  // adds a cache entry
-  return setEntry(service, entryKey, ctx.value, cache.ttl).then(() => next())
+  resolution
+    .then(() => next())
+    .catch(next)
```

Comment:
this could use ```.finally(() => next())```

Reply:
actually, in this case, its ok, if next() receives an argument it is handled as an error, so calling it empty keeps the train going, but catch() will pass an error object to next
-----------------
humm... one thing where you are right is that if line 314 produces an error, then catch will execute, and call next again with that error. 

probably store if there was an error and on finally call next with the error value or undefined as its default... prob I should do that :) thanks!

Reply:
I had a conversation with myself up there right? hahaah

Reply:
duh! http://bluebirdjs.com/docs/api/ascallback.html